### PR TITLE
Media support for Records

### PR DESCRIPTION
**Pull request by serge-gaia**
* New Media model
* API View, Serializers, Filters
* End point GET/POST /media   GET /media/{id}
* support for multipart upload only

TODO: base64 upload support

### CODE REVIEW

File: biosys/apps/main/models.py [hunk 1]
```diff
@@ -479,15 +479,28 @@ class Meta:
...
+        logger.exception('Error while building the media file name')
+        return 'unknown/{}'.format(filename)
+
+
 @python_2_unicode_compatible
-class DatasetFile(models.Model):
-    file = models.FileField(upload_to='%Y/%m/%d')
-    uploaded_date = models.DateTimeField(auto_now_add=True)
-    uploaded_by = models.ForeignKey(settings.AUTH_USER_MODEL, null=True, blank=True, on_delete=models.SET_NULL)
-    dataset = models.ForeignKey(Dataset, blank=False, null=True, on_delete=models.CASCADE)
```
Comment (serge-gaia) [inline biosys/apps/main/models.py:487]: Deleted this model. Never used.

File: biosys/apps/main/models.py [hunk 2]
```diff
@@ -479,15 +479,28 @@ class Meta:
         ordering = ['id']
 
 
+def media_path(instance, filename):
```
Comment (tony-gaia) [inline biosys/apps/main/models.py]: is this the right place for this function? Also, maybe call it ```get_media_path()```

Comment (serge-gaia) [reply biosys/apps/main/models.py]: Well it is used only in the Media model for the field `file`.
```
class Media(models.Model):
    file = models.FileField(upload_to=media_path)
```
This method should not be used.
I've added a comment
I've tried to put it as a static method inside the model but it didn't work.
https://docs.djangoproject.com/en/1.11/ref/models/fields/#filefield

I will change the name

**Comment by serge-gaia** [comment]
I have changed the name and added comments

**Pull request merged**